### PR TITLE
Feat/support beta extensions

### PR DESCRIPTION
**Pull request by jobala**
## Overview

Supports users switching between **beta** and **v1.0**

### Demo

**Selecting Graph Version**
image

**Show current graph version**
image

### Notes
The CLI will ship with both v1.0 and beta extensions then we use the graph version property to switch between beta and v1.0 extensions. The CLI loads v1.0 extensions by default.

## Testing Instructions

* Pull these changes
* Build and run the docker container, instructions are in the readme.
* Run `mg graph-version select` to select the graph version
* Run `mg graph-version show-current` to see the current graph version

## Files to review
* build_scripts/generate_extensions.py
* build_scripts/install_extensions.py
* msgraph/command_modules/profile/custom.py
* msgraph/cli/core/commands/client_factory.py

Closes #53

### CODE REVIEW

**Comment by MIchaelMainer** [comment]
With 4700+ files changed (most generated), are there specific things to look for? Also, which commits should I focus on?

**Comment by jobala** [comment]
@MIchaelMainer I have specified the files with changes in the PR description. Most of the work was done in commits made in Dec 1st and 2nd.

File: msgraph/command_modules/profile/custom.py [hunk 1]
```diff
@@ -76,6 +77,24 @@ def update_cloud(cloud: str, name=None, graph_endpoint=None, azure_ad_endpoint=N
...
+    selected = prompt_choice_list('Select graph version', supported_versions)
+    selected_version = supported_versions[selected]
+
+    profile.update({'version': selected_version})
+    write_profile(profile)
+
+    print(f'Version {selected_version} selected')
+
+
+def show_version():
```
Comment (MIchaelMainer) [inline msgraph/command_modules/profile/custom.py]: Where is show version used?

**Review by MIchaelMainer — COMMENTED** [review]
`sdvdiff 875770a HEAD build_scripts/generate_extensions.py build_scripts/install_extensions.py msgraph/command_modules/profile/custom.py msgraph/cli/core/commands/client_factory.py msgraph/cli/core/constants.py msgraph/cli/core/command_loaders/main.py msgraph/cli/core/__init__.py msgraph/command_modules/profile/__init__.py msgraph/command_modules/profile/_help.py`

File: msgraph/command_modules/profile/custom.py (hunk 1 above)
Comment (jobala) [reply msgraph/command_modules/profile/custom.py]: You're looking at an outdated file. Not sure why Github has not updated the file with the latest changes. The latest version of the file doesn't have show_version.

Comment (jobala) [reply msgraph/command_modules/profile/custom.py]: Can you try pulling and check if you see show_version?

Comment (MIchaelMainer) [reply msgraph/command_modules/profile/custom.py]: I'm at the head of the branch. I still see it.

Comment (jobala) [reply msgraph/command_modules/profile/custom.py]: Ah sorry, I had confused this comment and thought this  comment was made on the other the other PR -- #55  --  where I removed `show_version`.  

`show_version` is used in `msgraph/command_modules/profile/__init__.py`

**Review by MIchaelMainer — APPROVED** [review]

**Pull request merged**